packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -667,19 +674,19 @@ function useRestoreFocus(scopeRef: RefObject<Element[]>, restoreFocus?: boolean,
         && nodeToRestore
         && (
           // eslint-disable-next-line react-hooks/exhaustive-deps
-          isElementInScope(document.activeElement, scopeRef.current)
-          || (document.activeElement === document.body && shouldRestoreFocus(scopeRef))
+          isElementInScope(ownerDocument.activeElement, scopeRef.current)
+          || (ownerDocument.activeElement === ownerDocument.body && shouldRestoreFocus(scopeRef))
         )
       ) {
         // freeze the focusScopeTree so it persists after the raf, otherwise during unmount nodes are removed from it
         let clonedTree = focusScopeTree.clone();
         requestAnimationFrame(() => {
           // Only restore focus if we've lost focus to the body, the alternative is that focus has been purposefully moved elsewhere
-          if (document.activeElement === document.body) {
+          if (ownerDocument.activeElement === ownerDocument.body) {
             // look up the tree starting with our scope to find a nodeToRestore still in the DOM
             let treeNode = clonedTree.getTreeNode(scopeRef);
             while (treeNode) {
-              if (treeNode.nodeToRestore && document.body.contains(treeNode.nodeToRestore)) {
+              if (treeNode.nodeToRestore && ownerDocument.body.contains(treeNode.nodeToRestore)) {
```

Comment:
@snowystinger https://github.com/adobe/react-spectrum/pull/5449/files#diff-27278c0c2f69b8aa04b9d1fe837301b8e0f8a2fa216a75e70376d3f4e7f33e25R167 would be the test case -> the activeElement should ideally be the data-testid `outside` instead. It's because the `nodeToRestore` is not in the current `ownerDocument.body` since it's in another window object. Was looking at removing the `ownerDocument.body.contains(treeNode.nodeToRestore)` check but not sure what consequences it may have.

Reply:
This checks that the nodeToRestore is actually in the rendered DOM. It can happen that the nodeToRestore was removed at the same time as the activeElement, in which case we have to look higher up in our focus scope tree.

If you run all the tests, then you'll see some failures if you remove that statement. It's not directly tested in the FocusScope tests. It probably should be, but it's covered by real interactions with other components, so I'm not too worried about missing it here.

The nodeToRestore could be in the ownerDocument... or it might be in a different iframe document... so the better thing to do would probably be
```
if (treeNode.nodeToRestore && treeNode.nodeToRestore.isConnected) {
```
Caveat to that, I have no idea how that works in an iframe. I drafted a test for it, it seems to work. Would be good for you to double check it.
```
it('should not try to restore focus to an unmounted node inside an iframe', async function () {
      function Test({show, showInner}) {
        return (
          <div>
            <input data-testid="outside" />
            <IframeExample>
              {show &&
                <>
                  <FocusScope restoreFocus>
                    <input data-testid="iframe-outer" />
                    {showInner &&
                      <FocusScope restoreFocus>
                        <input data-testid="input1" />
                        <input data-testid="input2" />
                        <input data-testid="input3" />
                      </FocusScope>
                    }
                  </FocusScope>
                </>
              }
            </IframeExample>
          </div>
        );
      }

      let {getByTestId, rerender} = render(<Test />);

      let outside = getByTestId('outside');
      act(() => {outside.focus();});

      rerender(<Test show />);

      const iframeDocument = iframe.contentWindow.document;
      let iframeOuter = iframeDocument.querySelector('input[data-testid="iframe-outer"]');
      act(() => {iframeOuter.focus();});
      // Render inner focusScope after focus has moved to new spot, otherwise nodeToRestore will just be outer for both scopes, it's a little contrived, but is mostly how the real world works with restoring focusscopes being modals or something where focus is moved immediately in
      rerender(<Test show showInner />);
      const input1 = iframeDocument.querySelector('input[data-testid="input1"]');
      act(() => {input1.focus();});
      expect(iframeDocument.activeElement).toBe(input1);

      rerender(<Test />);
      iframe.remove();

      act(() => {jest.runAllTimers();});

      expect(document.activeElement).toBe(outside);
    });
```